File: src/main/java/com/raoulvdberge/refinedstorage/api/network/grid/INetworkAwareGrid.java
package com.raoulvdberge.refinedstorage.api.network.grid;

import com.raoulvdberge.refinedstorage.api.network.INetwork;

import javax.annotation.Nullable;

/**
 * A grid that knows about a network.
 */
public interface INetworkAwareGrid extends IGrid {
    /**
     * @return the network, or null if no network is available
     */
    @Nullable
    INetwork getNetwork();
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/network/grid/INetworkAwareGrid.kt
package com.refinedmods.refinedstorage.api.network.grid

import com.refinedmods.refinedstorage.api.network.INetwork


/**
 * A grid that knows about a network.
 */
interface INetworkAwareGrid : IGrid {
    /**
     * @return the network, or null if no network is available
     */
    val network: INetwork?
}